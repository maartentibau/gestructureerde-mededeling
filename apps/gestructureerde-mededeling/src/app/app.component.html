<ogm-navigation></ogm-navigation>
<div class="main">
  <div class="content">
    <router-outlet></router-outlet>
    <mat-card class="wiki">
      <mat-card-content>
        <p>
          Een overschrijving met gestructureerde mededeling of OGM is een combinatie van drie groepen van drie, vier en
          vijf cijfers gescheiden door een schuine streep.
        </p>
        <p>
          Deze mededeling wordt in BelgiÃ« vaak gebruikt om overschrijvingen automatisch te kunnen laten verwerken. Zo
          weet het computersysteem van de ontvanger onmiddellijk welke factuur betaald wordt. Op deze manier is er geen
          personeel nodig om te gaan kijken welke rekeningen vereffend werden.
        </p>
        <p>
          De eerste tien cijfers zijn bijvoorbeeld een klantennummer of een factuurnummer. De laatste twee cijfers
          vormen het controlegetal dat verkregen wordt door van de voorgaande tien cijfers de rest bij Euclidische
          deling door 97 te berekenen (modulo 97).
          <em>Uitzondering: Indien de rest 0 bedraagt, dan wordt als controlegetal 97 gebruikt.</em>
        </p>
        <p>Voor en achter de cijfers worden drie plussen (+++) of sterretjes (***) gezet.</p>
        <p>
          Bron:
          <a href="https://nl.wikipedia.org/wiki/Gestructureerde_mededeling" target="_blank" title="Source reference"
            >[wikipedia]</a
          >
        </p>
      </mat-card-content>
    </mat-card>
  </div>
  <div class="copyright">
    <small
      >Developed with <fa-icon [icon]="['fas', 'heart']" [ngStyle]="{ color: 'red' }"></fa-icon> by
      <a href="https://twitter.com/maartentibau" target="_blank">Maarten Tibau</a>
      <span>&bull;</span>
      <a href="https://www.buymeacoffee.com/maartentibau" target="_blank"
        ><img src="assets/images/buy-me-a-coffee.svg" height="13" /> Buy me a coffee</a
      >
      <span>&bull;</span> Source code on
      <a href="https://github.com/maartentibau/gestructureerde-mededeling" target="_blank"
        ><fa-icon [icon]="['fab', 'github']" [ngStyle]="{ color: 'black' }"></fa-icon> GitHub</a
      >
    </small>
  </div>
</div>
