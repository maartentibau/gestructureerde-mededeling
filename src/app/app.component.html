<ogm-navigation></ogm-navigation>
<div class="main">
  <div class="content">
    <router-outlet></router-outlet>
    <mat-card class="wiki">
      <mat-card-content>
        <p>
          Een overschrijving met gestructureerde mededeling of OGM is een combinatie van drie groepen van drie, vier en
          vijf cijfers gescheiden door een schuine streep.
        </p>
        <p>
          Deze mededeling wordt in BelgiÃ« vaak gebruikt om overschrijvingen automatisch te kunnen laten verwerken. Zo
          weet het computersysteem van de ontvanger onmiddellijk welke factuur betaald wordt. Op deze manier is er geen
          personeel nodig om te gaan kijken welke rekeningen vereffend werden.
        </p>
        <p>
          De eerste tien cijfers zijn bijvoorbeeld een klantennummer of een factuurnummer. De laatste twee cijfers
          vormen het controlegetal dat verkregen wordt door van de voorgaande tien cijfers de rest bij Euclidische
          deling door 97 te berekenen (modulo 97).
          <em>Uitzondering: Indien de rest 0 bedraagt, dan wordt als controlegetal 97 gebruikt.</em>
        </p>
        <p>Voor en achter de cijfers worden drie plussen (+++) of sterretjes (***) gezet.</p>
        <p>
          Bron:
          <a
            href="https://nl.wikipedia.org/wiki/Gestructureerde_mededeling"
            target="_blank"
            title="Source reference"
            rel="noopener noreferrer"
            >[wikipedia]</a
          >
        </p>
      </mat-card-content>
    </mat-card>
  </div>
  <div class="copyright">
    <small
      >Developed with <fa-icon [icon]="['fas', 'heart']" [ngStyle]="{ color: 'red' }"></fa-icon> by
      <a
        href="https://x.com/maartentibau"
        target="_blank"
        rel="noopener noreferrer"
        aria-label="Head over to my profile on X (Twitter)"
        >Maarten Tibau</a
      >
      <span>&bull;</span>
      <a
        href="https://www.buymeacoffee.com/maartentibau"
        target="_blank"
        rel="noopener noreferrer"
        aria-label="Buy me a coffee if you like to support me"
        ><img alt="Buy me a coffee logo" src="assets/images/buy-me-a-coffee.svg" height="13" /> Buy me a coffee</a
      >
      <span>&bull;</span> Source code on
      <a
        href="https://github.com/maartentibau/gestructureerde-mededeling"
        target="_blank"
        rel="noopener noreferrer"
        aria-label="You can find the source code of this project on Github"
        ><fa-icon [icon]="['fab', 'github']" [ngStyle]="{ color: 'black' }"></fa-icon> GitHub</a
      >
    </small>
  </div>
</div>
