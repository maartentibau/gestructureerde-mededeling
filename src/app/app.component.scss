@use '../assets/styles/screen-media-queries' as v;

.content {
  flex: 0 1 auto;
  max-width: 1920px;
  margin: 0 auto;
  overflow-x: hidden;
  overflow-y: auto;
  padding: 2rem;

  @media #{v.$mat-lt-sm} {
    padding: 0.7rem;
  }
}

.copyright {
  text-align: center;
  color: #595959;

  span {
    padding: 0 0.5rem;
  }

  a {
    color: #595959;
    text-decoration: none;
  }
}

mat-card.wiki {
  border-color: #efe897;
  background-color: #fffcd6;
  font-family: 'Lucida Console', Monaco, monospace;
  color: #5b5634;

  mat-card-content {
    font-size: 85%;

    .source-link {
      margin-left: 7px;
    }
  }
}
