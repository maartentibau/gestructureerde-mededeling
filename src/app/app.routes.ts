import { Route } from '@angular/router';

export const appRoutes: Route[] = [
  { path: '', loadComponent: () => import('./components/main/main.component').then((mod) => mod.MainComponent) },
  {
    path: 'create',
    loadComponent: () => import('./components/create/create.component').then((mod) => mod.CreateComponent),
  },
  {
    path: 'generate',
    loadComponent: () => import('./components/generate/generate.component').then((mod) => mod.GenerateComponent),
  },
  {
    path: 'validate',
    loadComponent: () => import('./components/validate/validate.component').then((mod) => mod.ValidateComponent),
  },
  {
    path: '**',
    loadComponent: () => import('./components/generate/generate.component').then((mod) => mod.GenerateComponent),
  },
];
