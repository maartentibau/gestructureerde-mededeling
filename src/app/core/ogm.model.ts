export interface OgmData {
  number: string;
  numberFormat: string;
}
