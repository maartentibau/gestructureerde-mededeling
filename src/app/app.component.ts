import { NgStyle } from '@angular/common';
import { Component, inject } from '@angular/core';
import { MatCardModule } from '@angular/material/card';
import { RouterOutlet } from '@angular/router';
import { FaConfig, FaIconLibrary, FontAwesomeModule } from '@fortawesome/angular-fontawesome';
import { faGithub } from '@fortawesome/free-brands-svg-icons';
import { faHeart } from '@fortawesome/free-solid-svg-icons';
import { NavigationComponent } from './core/components/navigation/navigation.component';

@Component({
  standalone: true,
  imports: [NavigationComponent, MatCardModule, FontAwesomeModule, NgStyle, RouterOutlet],
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.scss'],
})
export class AppComponent {
  #faConfig: FaConfig = inject(FaConfig);
  #faIconLibrary: FaIconLibrary = inject(FaIconLibrary);

  constructor() {
    this.#faConfig.fixedWidth = true;
    this.#faIconLibrary.addIcons(faGithub, faHeart);
  }
}
