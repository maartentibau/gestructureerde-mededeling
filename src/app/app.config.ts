import { ApplicationConfig, provideZonelessChangeDetection } from '@angular/core';

import { provideAnimationsAsync } from '@angular/platform-browser/animations/async';
import { provideRouter, withEnabledBlockingInitialNavigation } from '@angular/router';
import { appRoutes } from './app.routes';

export const appConfig: ApplicationConfig = {
  providers: [
    provideZonelessChangeDetection(),
    provideRouter(appRoutes, withEnabledBlockingInitialNavigation()),
    provideAnimationsAsync(),
  ],
};
