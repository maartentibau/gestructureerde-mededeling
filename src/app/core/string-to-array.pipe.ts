import { Pipe, PipeTransform } from '@angular/core';

@Pipe({
  standalone: true,
  name: 'stringToArray',
})
export class StringToArrayPipe implements PipeTransform {
  transform(value: number | string | null | undefined): string[] {
    if ((value && typeof value === 'string') || typeof value === 'number') {
      return Array.from(String(value));
    }

    return [];
  }
}
