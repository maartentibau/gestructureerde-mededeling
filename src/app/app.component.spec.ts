import { JsonPipe, NgStyle } from '@angular/common';
import { Component, Input } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { MatCardModule } from '@angular/material/card';
import { RouterTestingModule } from '@angular/router/testing';
import { FaConfig, FaIconLibrary } from '@fortawesome/angular-fontawesome';
import { AppComponent } from './app.component';

@Component({
  standalone: true,
  selector: 'ogm-navigation',
  template: '',
})
class MockNavigationComponent {}

@Component({
  standalone: true,
  imports: [JsonPipe],
  selector: 'fa-icon',
  template: ` <div>{{ icon | json }}</div> `,
})
class MockFaIconComponent {
  @Input() icon: string | string[] | undefined;
}

describe('AppComponent', () => {
  let component: AppComponent;
  let fixture: ComponentFixture<AppComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [AppComponent],
      providers: [
        { provide: FaConfig, useValue: { fixedWidth: null } },
        { provide: FaIconLibrary, useValue: { addIcons: jest.fn() } },
      ],
    })
      .overrideComponent(AppComponent, {
        set: { imports: [MockNavigationComponent, RouterTestingModule, MatCardModule, MockFaIconComponent, NgStyle] },
      })
      .compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(AppComponent);
    component = fixture.componentInstance;
  });

  it('should create the app', () => {
    expect(component).toBeTruthy();
  });

  describe('rendering component', () => {
    it('should match snapshot', () => {
      // act
      fixture.detectChanges();

      // check
      expect(fixture).toMatchSnapshot();
    });
  });
});
