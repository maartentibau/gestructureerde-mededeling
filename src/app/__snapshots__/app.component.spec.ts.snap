// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent rendering component should match snapshot 1`] = `
<ogm-app-root>
  <ogm-navigation /><div
    class="main"
  >
    <div
      class="content"
    >
      <router-outlet />
      <mat-card
        class="mat-mdc-card mdc-card wiki"
      >
        <mat-card-content
          class="mat-mdc-card-content"
        >
          <p>
             Een overschrijving met gestructureerde mededeling of OGM is een combinatie van drie groepen van drie, vier en vijf cijfers gescheiden door een schuine streep. 
          </p>
          <p>
             Deze mededeling wordt in België vaak gebruikt om overschrijvingen automatisch te kunnen laten verwerken. Zo weet het computersysteem van de ontvanger onmiddellijk welke factuur betaald wordt. Op deze manier is er geen personeel nodig om te gaan kijken welke rekeningen vereffend werden. 
          </p>
          <p>
             De eerste tien cijfers zijn bijvoorbeeld een klantennummer of een factuurnummer. De laatste twee cijfers vormen het controlegetal dat verkregen wordt door van de voorgaande tien cijfers de rest bij Euclidische deling door 97 te berekenen (modulo 97). 
            <em>
              Uitzondering: Indien de rest 0 bedraagt, dan wordt als controlegetal 97 gebruikt.
            </em>
          </p>
          <p>
            Voor en achter de cijfers worden drie plussen (+++) of sterretjes (***) gezet.
          </p>
          <p>
             Bron: 
            <a
              href="https://nl.wikipedia.org/wiki/Gestructureerde_mededeling"
              rel="noopener noreferrer"
              target="_blank"
              title="Source reference"
            >
              [wikipedia]
            </a>
          </p>
        </mat-card-content>
      </mat-card>
    </div>
    <div
      class="copyright"
    >
      <small>
        Developed with 
        <fa-icon
          style="color: red;"
        >
          <div>
            [
    "fas",
    "heart"
  ]
          </div>
        </fa-icon>
         by 
        <a
          href="https://twitter.com/maartentibau"
          rel="noopener noreferrer"
          target="_blank"
        >
          Maarten Tibau
        </a>
        <span>
          •
        </span>
        <a
          href="https://www.buymeacoffee.com/maartentibau"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            height="13"
            src="assets/images/buy-me-a-coffee.svg"
          />
           Buy me a coffee
        </a>
        <span>
          •
        </span>
         Source code on 
        <a
          href="https://github.com/maartentibau/gestructureerde-mededeling"
          rel="noopener noreferrer"
          target="_blank"
        >
          <fa-icon
            style="color: black;"
          >
            <div>
              [
    "fab",
    "github"
  ]
            </div>
          </fa-icon>
           GitHub
        </a>
      </small>
    </div>
  </div>
</ogm-app-root>
`;
